src/shape.cpp

```diff
@@ -175,12 +175,35 @@ bool Dimension::isAny() const {
            (this->minimum == DYNAMIC_DIMENSION);
 }
 
+// TODO decide during DAG task how it should look like & potentially fix
+bool Dimension::fitsInto(const Dimension& next) const {
+    if (next.isAny()) {
+        return true;
+    }
+    if (isStatic()) {
+        bool b = next.match(getStaticValue());
+        return b;
+    }
+    if (next.isStatic()) {
+        return this->match(next.getStaticValue());
+    }
+    // both are dynamic
+    if (next.getMinValue() > getMaxValue()) {
+        return false;
+    }
+    if (next.getMaxValue() > getMinValue()) {
+        return false;
+    }
+    return true;
+}
+
 bool Dimension::match(dimension_value_t value) const {
     if (isAny()) {
         return true;
     }
     if (isStatic()) {
-        return getStaticValue() == value;
+        return (getStaticValue() == value) ||
+               (getStaticValue() == 0);
```

Comment:
This will not work for models with dynamic shape of range [0~X]

Reply:
In this part of code we have StaticValue. But anyway it all comes down to handlin 0 which was dynamic in the previous releases of DAG. That's why I think it would be good to clean this up & everywhere internally have only -1 as dynamic, and then just translate all "external" coming "0" into -1.

Reply:
Yes, we have static shape here and we accept any argument if it is equal to 0. What if model has shape [5, 10, 0, 20] and we pass [5, 10, 100000, 20])? This should not match

Reply:
[5, 10, 0, 20] i doubt that there can be such tensor/model input. Anyway this still somewhat spilled DAG '0' treated as dynamic dimension into DImension class so I worked around that by using different call directly from PipelineDefinition